### ISSUE TITLE
`is` miscompilation

### ISSUE THREAD

**Issue by nullchinchilla** [mel-project/melodeon#5]
```
def dol(x: Nat) = [x]
def foo(x: Nat) =
	let y = dol(x) in
	if y is [Nat] then
		y[0]
	else
		y
---
foo(2)
```
should evaluate to `2` but instead evaluates to `[2]`, indicating that the `y is [Nat]` is miscompiled.

**Comment by nullchinchilla** [comment]
Current codegen produces

```scheme
(fn foo (x)
     (let (y (dol x))
       (if (and (and (= 2 (typeof y)) (= 1 (v-len y)))
                (= 0 (typeof y)))
           (v-get y 0)
           y)))
```

which is obvious wrong, since it's checking that the *entire vector* is a Nat. This is probably just a bug in the is codegen again.

**Comment by nullchinchilla** [comment]
fixed in [98bbb48](https://github.com/themeliolabs/melodeon/commit/98bbb48e3438b860919aeffc0c83e22a463ca345)
